src/gcodeExport.cpp

```diff
@@ -904,11 +905,13 @@ void GCodeExport::writeRetraction(const RetractionConfig& config, bool force, bo
         }
         if (!force && config.retraction_count_max <= 0)
         {
+            writeComment("Retraction cancelled - retraction max <= 0");
             return;
         }
         if (!force && extruded_volume_at_previous_n_retractions.size() == config.retraction_count_max
             && current_extruded_volume < extruded_volume_at_previous_n_retractions.back() + config.retraction_extrusion_window * extr_attr.filament_area) 
         {
+            writeComment("Retraction cancelled - retraction max exceeded");
```

Comment:
I would like to add some more detail about how far back the n'th retraction was, but my c++ is not up to it.